src/amber/router/session/redis_store.cr

```diff
@@ -0,0 +1,81 @@
+require "redis"
+
+module Amber::Router::Session
+  class RedisStore < AbstractStore
+    getter store : Redis
+    property expires : Int32
+    property key : String
+    property session_id : String
+    property cookies : Amber::Router::Cookies::Store
+    property id : String
+
+    def initialize(@store, @cookies, @key, @expires = 120)
+      @id = current_session
+      @session_id = "#{@key}:#{@id}"
+    end
+
+    def destroy
+      @store.del(session_id)
+    end
+
+    def [](key)
+      @store.hget(session_id, key)
+    end
+
+    def []?(key)
+      key?(key)
+    end
+
+    def []=(key, value)
+      @store.hset(session_id, key, value)
+    end
+
+    def key?(key)
+      @store.hexists(session_id, key) == 1 ? true : false
+    end
+
+    def keys
+      @store.hkeys(session_id)
+    end
+
+    def values
+      @store.hvals(session_id)
+    end
+
+    def to_h
+      @store.hmget(session_id).each_slice(2).to_h
+    end
+
+    def update(hash : Hash(String, String))
+      @store.hmset(session_id, hash)
+    end
+
+    def delete(key)
+      @store.hdel(session_id, key) if key?(key)
+    end
+
+    def fetch(key, default = nil)
+      return self[key] if key?(key)
+      (self[key] = default)
```

Comment:
If a key doesn't exist why are we setting it with a value of default? Shouldn't it continue to not exist and just return default? Like with in the stdlib hash: https://github.com/crystal-lang/crystal/blob/d1f8c42f82d3e6491cb7b3b536984c19a51b94ba/src/hash.cr#L137